### ISSUE TITLE
Rule to disallow conditional test definition

### ISSUE THREAD

**Issue by m-burst** [m-burst/flake8-pytest-style#29]
# Rule request

## Description

Test definition should not be placed in a top-level `if` block.  `pytest.mark.skipif()` should be used instead, if necessary.

Bad code:

```py
if sys.platform == 'win32':
    def test_on_windows_only():
        ...
```

Good code:
```py
@pytest.mark.skipif(sys.platform != 'win32', reason='windows only')
def test_on_windows_only():
    ...
```

## Rationale

* for consistency
* to make such conditionally skipped tests appear in the report
